components/script/script_thread.rs

```diff
@@ -3567,11 +3589,14 @@ impl ScriptThread {
                 // We might have to reset the anchor state
                 if !state_already_changed {
                     if let Some(target) = prev_mouse_over_target {
-                        if let Some(_) = target
-                            .upcast::<Node>()
-                            .inclusive_ancestors(ShadowIncluding::No)
-                            .filter_map(DomRoot::downcast::<HTMLAnchorElement>)
-                            .next()
+                        if Some(
+                            target
+                                .upcast::<Node>()
+                                .inclusive_ancestors(ShadowIncluding::No)
+                                .filter_map(DomRoot::downcast::<HTMLAnchorElement>)
+                                .next(),
+                        )
+                        .is_some()
```

Comment:
Why are you making this change? `Some(something).is_some()` always returns true, so this change breaks this code.